src/api/middleware/company.js

```diff
@@ -85,6 +85,69 @@ export const profileComplete = async (req, res, next) => {
     return next();
 };
 
+export const restrictedAccess = (owner) => async (req, res, next) => {
+    const company = await (new CompanyService()).findById(owner, true);
+    let error = {};
+
+    if (company.isDisabled) {
+        if (req.params?.companyId === req.user.company) {
+            error = new APIError(
+                HTTPStatus.OK,
+                ErrorTypes.VALIDATION_ERROR,
+                ValidationReasons.COMPANY_DISABLED,
+                { company: company }
+            );
+        } else {
+            error = new APIError(
+                HTTPStatus.FORBIDDEN,
+                ErrorTypes.FORBIDDEN,
+                ValidationReasons.NOT_FOUND
+            );
+        }
+    }
+
+    if (company.isBlocked) {
+        if (req.params?.companyId === req.user.company) {
+            error = new APIError(
+                HTTPStatus.OK,
+                ErrorTypes.VALIDATION_ERROR,
+                ValidationReasons.COMPANY_BLOCKED,
+                { company: company }
+            );
+        } else {
+            error = new APIError(
+                HTTPStatus.FORBIDDEN,
+                ErrorTypes.FORBIDDEN,
+                ValidationReasons.NOT_FOUND
+            );
+        }
+    }
```

Comment:
There is a lot of code that is repeated, I would suggest doing something like this:

```suggestion
    if (req.params?.companyId === req.user.company) {
        error = new APIError(
            HTTPStatus.OK,
            ErrorTypes.VALIDATION_ERROR,
            company.isBlocked ? ValidationReasons.COMPANY_BLOCKED : company.isDisabled ? ValidationReasons.COMPANY_DISABLED : ValidationReasons.UNKNONW, // this here can be any other validation reason, put UNKOWN as an example
            { company: company }
        );
    } else {
        error = new APIError(
            HTTPStatus.FORBIDDEN,
            ErrorTypes.FORBIDDEN,
            ValidationReasons.NOT_FOUND
        );
    }
```

This will give priority to the blocked information so we should think more about how to handle this.

Reply:
The chain is a bit confusing, at least put brackets to differentiate a bit:
```js
company.isBlocked ? ValidationReasons.COMPANY_BLOCKED : 
  (company.isDisabled ? ValidationReasons.COMPANY_DISABLED : ValidationReasons.UNKNONW)
```